x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form.tsx

```diff
@@ -77,6 +85,70 @@ export const PipelineForm: React.FunctionComponent<Props> = ({
     onSubmit: handleSave,
   });
 
+  const renderProcessorsEditorSection = () => {
+    if (showJsonProcessorsEditor) {
+      return (
+        <FormRow
+          title={
+            <FormattedMessage
+              id="xpack.ingestPipelines.form.processorsFieldTitle"
+              defaultMessage="Processors"
+            />
+          }
+          description={
+            <FormattedMessage
+              id="xpack.ingestPipelines.form.processorsFieldDescription"
+              defaultMessage="The processors used to pre-process documents before indexing. {learnMoreLink}"
+              values={{
+                learnMoreLink: (
+                  <EuiLink href={services.documentation.getProcessorsUrl()} target="_blank">
+                    {i18n.translate('xpack.ingestPipelines.form.processorsDocumentionLink', {
+                      defaultMessage: 'Learn more.',
+                    })}
+                  </EuiLink>
+                ),
+              }}
+            />
+          }
+        >
+          <UseField
+            path="processors"
+            component={JsonEditorField}
+            componentProps={{
+              ['data-test-subj']: 'processorsField',
+              euiCodeEditorProps: {
+                height: '300px',
+                'aria-label': i18n.translate(
+                  'xpack.ingestPipelines.form.processorsFieldAriaLabel',
+                  {
+                    defaultMessage: 'Processors JSON editor',
+                  }
+                ),
+              },
+            }}
+          />
+        </FormRow>
+      );
+    }
+
+    const formProcessorValue = form.getFields().processors?.value ?? defaultValue?.processors;
```

Comment:
We should use the `<FormDataProvider>` to access the form data.

So this line and the one below are not needed. Neither the `processors` prop on the PipelineProcessorsEditor component. Then inside that component you wrap the content like this

```js
<FormDataProvider pathsToWatch="processors">
  {({ processors }) => {
    // use the data
  }}
</FormDataProvider>
```

Reply:
> Neither the processors prop on the PipelineProcessorsEditor component.

It seems a bit cleaner to me, if we want to the pipeline processor editor to be reusable, to still have the processors prop and not rely on the form data provider being there. Additionally, the component also does some pre and post processing to the raw processors data.

Reply:
In that case, you wrap the processor with it 😊 

```js
<FormDataProvider pathsToWatch="processors">
  {({ processors }) => {
    // Do anything with the raw data
    return (
      <PipelineProcessorsEditor
        stateReaderRef={ref => setProcessorStateReaderRef(ref)}
        processors={processors}
      />
    );
  }}
</FormDataProvider>
```

With the above, you will be updated each time the processors value changes, and it is the recommended way to access the data.